ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/csv/DetailedObjectsPhysicalView.java

```diff
@@ -15,63 +15,116 @@
 
 package com.spectralogic.ds3cli.views.csv;
 
+import com.google.common.base.Function;
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.spectralogic.ds3cli.View;
 import com.spectralogic.ds3cli.models.GetDetailedObjectsResult;
 import com.spectralogic.ds3client.models.*;
 import com.spectralogic.ds3client.utils.Guard;
 
-import java.util.ArrayList;
+import javax.annotation.Nullable;
 
 import static com.spectralogic.ds3cli.util.Utils.*;
+import static com.spectralogic.ds3cli.views.csv.CsvView.DATE_FORMAT;
 
-public class DetailedObjectsPhysicalView extends CsvView<GetDetailedObjectsResult> {
-    private ImmutableList<DetailedS3Object> objects;
+public class DetailedObjectsPhysicalView implements View<GetDetailedObjectsResult> {
 
     @Override
     public String render(final GetDetailedObjectsResult obj) {
-        if (obj == null || (obj.getObjIterator() == null) || !obj.getObjIterator().iterator().hasNext()) {
+        if (obj == null || obj.getObjIterator() == null || Iterables.isEmpty(obj.getObjIterator())) {
             return "No objects returned";
         }
 
-        objects = ImmutableList.copyOf(obj.getObjIterator());
-        initTable(ImmutableList.of("Name", "Bucket", "Owner", "Size", "Type", "Creation Date", "Barcode", "State"));
+        final ImmutableList<String> headers = ImmutableList.of("Name", "Bucket", "Owner", "Size", "Type", "Creation Date", "Barcode", "State");
 
-        return renderTable();
+        final FluentIterable<DetailedTapeInfo> objects = FluentIterable.from(obj.getObjIterator())
+                .filter(new Predicate<DetailedS3Object>() {
+                    @Override
+                    public boolean apply(@Nullable final DetailedS3Object input) {
+                        return input.getBlobs() != null && !Guard.isNullOrEmpty(input.getBlobs().getObjects());
+                    }
```

Comment:
Maybe a method on DetailedS3Object to have it indicate whether it has stuff to process?